src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -205,24 +237,34 @@ class FastSync(
       }
     }
 
-    private def updateTargetSyncState(state: FinalBlockProcessingResult, targetBlockHeader: BlockHeader): Unit = state match {
-      case ImportedLastBlock =>
-        if (targetBlockHeader.number - syncState.targetBlock.number <= syncConfig.maxTargetDifference) {
-          log.info(s"Current target block is fresh enough, starting state download")
-          if (syncState.targetBlock.stateRoot == ByteString(MerklePatriciaTrie.EmptyRootHash)) {
-            syncState = syncState.copy(pendingMptNodes = Seq())
+    private def updateTargetSyncState(state: FinalBlockProcessingResult, targetBlockHeader: BlockHeader): Unit =
+      state match {
+        case ImportedLastBlock =>
+          if (targetBlockHeader.number - syncState.targetBlock.number <= syncConfig.maxTargetDifference) {
+            log.info(s"Current target block is fresh enough, starting state download")
+            if (syncState.targetBlock.stateRoot == ByteString(MerklePatriciaTrie.EmptyRootHash)) {
```

Comment:
For now i will add comment here. Also it can be true, for any block number as long there were no transaction before. i.e
if block `1000` has empty root that means there mere not transaction up until this point.